### ISSUE TITLE
Test suite broken by unordered-containers 0.2.16.0

### ISSUE THREAD

**Issue by sternenseemann** [kapralVV/Unique#9]
Test suite tests precise ordering returned from unordered-containers which is not guaranteed.

```
Running 1 test suites...
Test suite HspecTest: RUNNING...

Data.List.Unique.isUnique
  isUnique: should return Nothing with empty list
  isUnique: should return Nothing with empty list (quickCheck)
    +++ OK, passed 100 tests.
  isUnique: returns (Just True) when element is unique
  isUnique: returns (Just False) when list has duplicate of element
  isUnique: is a lazy function, 'undefined test'
  isUnique: is a lazy function, 'undefined test'. Generates exeption
  isRepeated is a lazy function too, 'undefined test'
  isRepeated: is reverse function to isUnique
    +++ OK, passed 100 tests.
  isUnique: should return (Just ANY) when element is exist in the list
    +++ OK, passed 100 tests.
  isUnique: should return (Just False) when at least two elements are exist in the list
    - It checks laziness as well
      +++ OK, passed 100 tests.
  isUnique: should return Nothing when element is absent in the list
    +++ OK, passed 100 tests; 14 discarded.
  isUnique: and isRepeated should return Nothing when element is absent in the list
    +++ OK, passed 100 tests; 14 discarded.

Finished in 0.0040 seconds
12 examples, 0 failures

Data.List.Unique.sortUniq
  sortUniq: should return [] with empty list
  sortUniq: simple test
  sortUniq: multiple execution should return the same result
    +++ OK, passed 100 tests.
  sortUniq: the result has to be sorted
    +++ OK, passed 100 tests.
  sortUniq: elements should occur only once
    +++ OK, passed 100 tests.
  sortUniq: elements should occur only once #2
    +++ OK, passed 100 tests.
  sortUniq: check if it's correct by slow analog function
    +++ OK, passed 100 tests.
  sortUniq: check if it's correct by the Faster analog function :)
    +++ OK, passed 100 tests.

Finished in 0.0135 seconds
8 examples, 0 failures

Data.List.Unique.repeatedBy,repeated,unique
  repeatedBy: should return [] with empty list
  repeatedBy: simple test
  repeatedBy: returns [] when predicate (=< negative) 
    +++ OK, passed 100 tests; 109 discarded.
  repeatedBy: removes the duplicated elements like `sortUniq` function when predicate (>= negative)
    +++ OK, passed 100 tests; 109 discarded.
  repeatedBy: returns [] when preficate (== 0)
    +++ OK, passed 100 tests.
  repeatedBy: returns sorted result
    +++ OK, passed 100 tests; 119 discarded.
  repeatedBy: resulted elements should occur only once
    +++ OK, passed 100 tests; 119 discarded.
  unique: simple test
  repeated: simple test
  unique: multiple execution should return the same result
    +++ OK, passed 100 tests.
  repeated: multiple execution should return []
    +++ OK, passed 100 tests.
  repeated && unique: should return different result on the same non empty list
    +++ OK, passed 100 tests; 15 discarded.

Finished in 0.0134 seconds
12 examples, 0 failures

Data.List.Unique.complex
  complex: should return ([],[],[]) with empty list
  complex: simple test
  complex: returns the same result as `sortUniq`, `repeated`, `unique` but not sorted
    +++ OK, passed 100 tests.

Finished in 0.0044 seconds
3 examples, 0 failures

Data.List.Unique.allUnique
  allUnique: returns True with empty list
  allUnique: returns False when some element is not unique
  allUnique: returns True when list does not have duplicates
  allUnique: fails if list consist of duplicate elements and 'undefined' after them
  allUnique: Test the function using slower analog and random data
    +++ OK, passed 100 tests.

Finished in 0.0012 seconds
5 examples, 0 failures

Data.List.UniqueStrict.isUnique
  isUnique: should return Nothing with empty list
  isUnique: should return Nothing with empty list (quickCheck)
    +++ OK, passed 100 tests.
  isUnique: returns (Just True) when element is unique
  isUnique: returns (Just False) when list has duplicate of element
  isUnique: is NOT a lazy function, 'undefined test'
  isRepeated: is reverse function to isUnique
    +++ OK, passed 100 tests.
  isUnique: should return (Just ANY) when element is exist in the list
    +++ OK, passed 100 tests.
  isUnique: should return Nothing when element is absent in the list
    +++ OK, passed 100 tests; 14 discarded.
  isUnique: and isRepeated should return Nothing when element is absent in the list
    +++ OK, passed 100 tests; 14 discarded.

Finished in 0.0056 seconds
9 examples, 0 failures

Data.List.UniqueStrict.sortUniq
  sortUniq: should return [] with empty list
  sortUniq: simple test
  sortUniq: multiple execution should return the same result
    +++ OK, passed 100 tests.
  sortUniq: the result has to be sorted
    +++ OK, passed 100 tests.
  sortUniq: elements should occur only once
    +++ OK, passed 100 tests.
  sortUniq: elements should occur only once #2
    +++ OK, passed 100 tests.
  sortUniq: check if it's correct by slow analog function
    +++ OK, passed 100 tests.
  sortUniq: check if it's correct by the Faster analog function :)
    +++ OK, passed 100 tests.

Finished in 0.0077 seconds
8 examples, 0 failures

Data.List.UniqueStrict.repeatedBy,repeated,unique
  repeatedBy: should return [] with empty list
  repeatedBy: simple test
  repeatedBy: returns [] when predicate (=< negative) 
    +++ OK, passed 100 tests; 96 discarded.
  repeatedBy: removes the duplicated elements like `sortUniq` function when predicate (>= negative)
    +++ OK, passed 100 tests; 96 discarded.
  repeatedBy: returns [] when preficate (== 0)
    +++ OK, passed 100 tests.
  repeatedBy: returns sorted result
    +++ OK, passed 100 tests; 119 discarded.
  repeatedBy: resulted elements should occur only once
    +++ OK, passed 100 tests; 119 discarded.
  unique: simple test
  repeated: simple test
  unique: multiple execution should return the same result
    +++ OK, passed 100 tests.
  repeated: multiple execution should return []
    +++ OK, passed 100 tests.
  repeated && unique: should return different result on the same non empty list
    +++ OK, passed 100 tests; 19 discarded.

Finished in 0.0108 seconds
12 examples, 0 failures

Data.List.UniqueStrict.allUnique
  allUnique: returns True with empty list
  allUnique: returns False when some element is not unique
  allUnique: returns True when list does not have duplicates
  allUnique: fails if list consist of duplicate elements and 'undefined' after them
  allUnique: Test the function using slower analog and random data
    +++ OK, passed 100 tests.

Finished in 0.0015 seconds
5 examples, 0 failures

Data.List.UniqueUnsorted.isUnique
  isUnique: should return Nothing with empty list
  isUnique: should return Nothing with empty list (quickCheck)
    +++ OK, passed 100 tests.
  isUnique: returns (Just True) when element is unique
  isUnique: returns (Just False) when list has duplicate of element
  isUnique: is NOT a lazy function, 'undefined test'
  isRepeated: is reverse function to isUnique
    +++ OK, passed 100 tests.
  isUnique: should return (Just ANY) when element is exist in the list
    +++ OK, passed 100 tests.
  isUnique: should return Nothing when element is absent in the list
    +++ OK, passed 100 tests; 11 discarded.
  isUnique: and isRepeated should return Nothing when element is absent in the list
    +++ OK, passed 100 tests; 11 discarded.

Finished in 0.0042 seconds
9 examples, 0 failures

Data.List.UniqueUnsorted.removeDuplicates
  removeDuplicates: should return [] with empty list
  removeDuplicates: simple test FAILED [1]
  removeDuplicates: multiple execution should return the same result
    +++ OK, passed 100 tests.
  removeDuplicates: elements should occur only once
    +++ OK, passed 100 tests.
  removeDuplicates: elements should occur only once #2
    +++ OK, passed 100 tests.

Failures:

  tests/UniqueUnsorted/RemoveDuplicates.hs:18:5: 
  1) Data.List.UniqueUnsorted.removeDuplicates removeDuplicates: simple test
       expected: " abrfo"
        but got: " abfor"

  To rerun use: --match "/Data.List.UniqueUnsorted.removeDuplicates/removeDuplicates: simple test/"

Randomized with seed 1708031936

Finished in 0.0048 seconds
5 examples, 1 failure
Test suite HspecTest: FAIL
Test suite logged to: dist/test/Unique-0.4.7.9-HspecTest.log
0 of 1 test suites (0 of 1 test cases) passed.
```

**Comment by sternenseemann** [comment]
Problematic tests:

* "/Data.List.UniqueUnsorted.removeDuplicates/removeDuplicates: simple test/"
* "/Data.List.UniqueUnsorted.repeatedBy,repeated,unique/unique: simple test/"
* "/Data.List.UniqueUnsorted.repeatedBy,repeated,unique/repeatedBy: simple test/"

**Comment by Unisay** [comment]
FYI: [This PR](https://github.com/kapralVV/Unique/pull/10) fixes problematic tests.
